Python/bytecodes.c

```diff
@@ -1138,8 +1138,10 @@ dummy_func(
                 (Py_TYPE(receiver_o) == &PyGen_Type || Py_TYPE(receiver_o) == &PyCoro_Type) &&
                 ((PyGenObject *)receiver_o)->gi_frame_state < FRAME_EXECUTING)
             {
+                _PyInterpreterFrame *gen_frame;
                 PyGenObject *gen = (PyGenObject *)receiver_o;
-                _PyInterpreterFrame *gen_frame = &gen->gi_iframe;
+                Py_BEGIN_CRITICAL_SECTION(gen);
```

Comment:
What about _SEND_GEN_FRAME, YIELD_VALUE, _FOR_ITER_GEN_FRAME and RETURN_GENERATOR instructions?

Reply:
I'm not sure we need to mess with things like `YIELD_VALUE`. A thread can't yield another thread's generator, can it?